### DISCUSSION TITLE
for each support ？

### DISCUSSION THREAD

**Discussion by Another7** [microsoft/vscode-cpptools#7425] [Q&A] [answered]
~~~c++
int main()
{
    map<string, int> mymap;
    mymap["surrender"] = 1;
    for (auto [key, value] : mymap)
    {
        cout << key << "---" << value << endl;
    }
    return 0;
}
~~~
![image](https://user-images.githubusercontent.com/27720714/116021058-01b82b80-a67a-11eb-8d67-3e2d26d51fb8.png)
# Can compile successfully but prompt undefined identifier

**Comment by Colengms** [comment, marked as answer]
Hi @Another7 .  I do not get errors with the following code:

```cpp
#include <iostream>
#include <string>
#include <map>

using namespace std;

int main()
{
    map<string, int> mymap;
    mymap["surrender"] = 1;
    for (auto [key, value] : mymap)
    {
        cout << key << "---" << value << endl;
    }
    return 0;
}
```

I suspect you will need to provide a `cppStandard` setting of `c++17` or `c++20`, for this code to be considered valid.  If you have not set `cppStandard`, it would be the default used by the specified or detected compiler, which likely is not `c++17` or `c++20`.

**Reply by Another7** [reply]
Got it, Sorry for this question and thanks for you.
